samples/drivers/i2c_scanner/src/main.c

```diff
@@ -9,13 +9,19 @@
 #include <sys/printk.h>
 #include <device.h>
 #include <drivers/i2c.h>
+#include <generated_dts_board.h>
 
 #ifdef ARDUINO_I2C_LABEL
 #define I2C_DEV ARDUINO_I2C_LABEL
 #else
 #define I2C_DEV "I2C_0"
 #endif
 
+#define device_get_dt_binding_zdid(zdid) _CONCAT(&__device_, zdid)
+
+#define device_get_dt_binding(name, inst) \
+device_get_dt_binding_zdid(DT_INST_##inst##_##name##_ZDID) \
```

Comment:
avoid it how?

Reply:
first, what would be better?

Reply:
> avoid it how?

Exactly my question :) 

I understand the idea behind this patches, however it's changing how user will get the bindings. It could be done directly with a static string being the same as the dts node label ("SPI_0" for instance). That was simple.

Also doing it that way was closely down to the board the user is using. He probably does not know which controller runs on the SoC, he has a board that exposes "spi_0, spi_1, uart_0, uart_1, uart_2" and it is then direct: device_get_binding("UART_0"). One can relate it directly to the board. 

Now it would require the user to know the controller, and the instance he needs, exposing then the instance concept to the user at the same time then... not sure it's wise.

Looks like I am in line with @erwango here.

Reply:
Keeping the labels available under a default y kconfig would let users keep doing this for upstream samples but still let us get rid of the labels to save ROM when needed, agreed @tbursztyka ?

Reply:
When it comes to name, I think we want to get rid of them entirely. They are really useless.
(only maybe in debug mode, inside the actual driver, but nowhere else).

Instead we could generate BOARD_UART_0, BOARD_SPI_1, ... macros which in fact would hold the dev name (which is being build with the ZDID etc.. on this rfc). device_get_binding() would then call DEVICE_GET() with this, and since dts_device_extern.h already exposes the extern device pointers: done.

@galak That could be your solution to avoid fully deprecating device_get_binding(), though it would change it's signature (it would become a macro calling DEVICE_GET() macro).

Reply:
I expect several `device_get` APIs that would be new.  We can continue to support `device_get_binding()` via string.  I'd expect we'd have:

`device_get_by_dt_label(LABEL) [this is the node label, not the 'label' property]`
`device_get_by_dt_alias(ALIAS)`
`device_get_by_dt_path(PATH)`
`device_get_by_dt_inst(NAME, ID)` [not sure if this one is useful]

Reply:
ok sure, device_get_binding() would use device_get_by_dt_label then.

Reply:
>device_get_by_dt_inst(NAME, ID) [not sure if this one is useful]

If the NAME is type of the device and ID the index, then it would allow the same feature as suggested in #21709